test/adapter_test.rb

```diff
@@ -0,0 +1,58 @@
+require 'test_helper'
+
+class TestSemianAdapter < Minitest::Test
+  def setup
+    Semian.unregister_all_resources
+    # Consumers registered in other test files must be cleared
+    Semian.instance_variable_set(:@consumers, nil)
+    GC.enable
+    GC.start
+  end
+
+  def teardown
+    Semian.unregister_all_resources
+  end
+
+  def test_adapter_registers_consumer
+    assert_equal(true, Semian.resources.empty?)
+    assert_equal(true, Semian.consumers.empty?)
+    client = Semian::AdapterTestClient.new(quota: 0.5)
+    resource = client.semian_resource
+    assert_equal(resource, Semian.resources[client.semian_identifier])
+    assert_equal(client.inspect, Semian.consumers[client.semian_identifier].first.inspect)
```

Comment:
this was me being lazy and not writing a comparator. minitest was complaining that the inspect was identical but that it didn't have a comparator for Semian::AdapterTestClient. I can write one.